### PR TITLE
fix(sdk): turn `indent` option of `json.stringify` into keyword argument

### LINKED ISSUE #2246: `indent` option in `Json.stringify` needs to be a keyword argument and not a positional argument
### I tried this

```js
Json.stringify(foo, indent: 2);
```

### I expected this:

To work

### Instead, this happened

Apparently the `indent` option is a positional argument:

### Is there a workaround?

```js
Json.stringify(foo, 2);
```

### Component

SDK

### Wing Version

_No response_

### Wing Console Version

_No response_

### Node.js Version

_No response_

### Platform(s)

_No response_

### Anything else?

_No response_

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### PR DESCRIPTION
**Pull request by garysassano**
Closes #2246

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

**Comment by tsuf239** [comment]
Can you please add a test?

**Review by eladb — CHANGES_REQUESTED** [review]

File: libs/wingsdk/src/std/json.ts [hunk 1]
```diff
@@ -42,15 +46,18 @@ export class Json {
    * Formats Json as string
    *
    * (JSON.stringify($args$))
-   *
-   * @macro ((args) => { return JSON.stringify(args[0], null, args[1]) })([$args$])
+
+   * @macro ((args) => { return JSON.stringify(args[0], null, args[1].indent) })([$args$])
```
Comment (MarkMcCulloh) [inline libs/wingsdk/src/std/json.ts:50]:
```suggestion
   * @macro ((args) => { return JSON.stringify(args[0], null, args[1]?.indent) })([$args$])
```

File: libs/wingsdk/src/std/json.ts [hunk 2]
```diff
@@ -42,15 +46,18 @@ export class Json {
...
    *
    * @param json to format as string
    * @returns string representation of the Json
    */
-  public static stringify(json: Json | MutJson, indent?: number): string {
+  public static stringify(
+    json: Json | MutJson,
+    options: JsonStringifyOptions = { indent: 0 }
```
Comment (MarkMcCulloh) [inline libs/wingsdk/src/std/json.ts:57]: I don't think we can (or should) rely on a default option here

```suggestion
    options?: JsonStringifyOptions
```

File: libs/wingsdk/src/std/json.ts [hunk 3]
```diff
@@ -1,5 +1,9 @@
 import { Code, InflightClient } from "../core";
 
+export interface JsonStringifyOptions {
```
Comment (MarkMcCulloh) [inline libs/wingsdk/src/std/json.ts:3]: Missing docstrings

**Review by MarkMcCulloh — CHANGES_REQUESTED** [review]

File: libs/wingsdk/src/std/json.ts (hunk 2 above)
Comment (garysassano) [reply libs/wingsdk/src/std/json.ts:57]: Yeah, @Chriscbr confirmed that the compiler ignores that.

**Review by eladb — APPROVED** [review]
LGTM

File: libs/wingsdk/src/std/json.ts [hunk 4]
```diff
@@ -42,15 +49,18 @@ export class Json {
    * Formats Json as string
    *
    * (JSON.stringify($args$))
-   *
-   * @macro ((args) => { return JSON.stringify(args[0], null, args[1]) })([$args$])
+
```
Comment (MarkMcCulloh) [inline libs/wingsdk/src/std/json.ts:52]: extra newline

```suggestion
```

File: libs/wingsdk/src/std/json.ts [hunk 5]
```diff
@@ -42,15 +49,18 @@ export class Json {
    * Formats Json as string
    *
    * (JSON.stringify($args$))
```
Comment (MarkMcCulloh) [inline libs/wingsdk/src/std/json.ts:51]: I know it's not part of your change, but this macro-looking thing probably shouldn't be here

```suggestion
```

**Review by MarkMcCulloh — APPROVED** [review]

File: libs/wingsdk/src/std/json.ts [hunk 6]
```diff
@@ -1,5 +1,12 @@
 import { Code, InflightClient } from "../core";
 
+/**
+ * Options for stringify() method.
+ */
+export interface JsonStringifyOptions {
+  readonly indent: number;
```
Comment (MarkMcCulloh) [inline libs/wingsdk/src/std/json.ts:7]: Missing docstring on this property

**Pull request merged**